deeppavlov/core/commands/infer.py

```diff
@@ -38,66 +43,103 @@ def build_model(config: Union[str, Path, dict], mode: str = 'infer',
+    model = Chainer(model_config["in"], model_config["out"], model_config.get("in_y"))
 
-    for component_config in model_config['pipe']:
-        if load_trained and ('fit_on' in component_config or 'in_y' in component_config):
+    for component_config in model_config["pipe"]:
+        if load_trained and (
+            "fit_on" in component_config or "in_y" in component_config
+        ):
             try:
-                component_config['load_path'] = component_config['save_path']
+                component_config["load_path"] = component_config["save_path"]
             except KeyError:
-                log.warning('No "save_path" parameter for the {} component, so "load_path" will not be renewed'
-                            .format(component_config.get('class_name', component_config.get('ref', 'UNKNOWN'))))
+                log.warning(
+                    'No "save_path" parameter for the {} component, so "load_path" will not be renewed'.format(
+                        component_config.get(
+                            "class_name", component_config.get("ref", "UNKNOWN")
+                        )
+                    )
+                )
 
         component = from_params(component_config, mode=mode)
 
-        if 'id' in component_config:
-            model._components_dict[component_config['id']] = component
+        if "id" in component_config:
+            model._components_dict[component_config["id"]] = component
 
-        if 'in' in component_config:
-            c_in = component_config['in']
-            c_out = component_config['out']
-            in_y = component_config.get('in_y', None)
-            main = component_config.get('main', False)
+        if "in" in component_config:
+            c_in = component_config["in"]
+            c_out = component_config["out"]
+            in_y = component_config.get("in_y", None)
+            main = component_config.get("main", False)
             model.append(component, c_in, c_out, in_y, main)
 
     return model
 
 
+def end_repl_mode(function: Callable[..., Any]) -> Callable[..., Any]:
+    """Decorator for processing ctrl-c, ctrl-d pressing."""
+
+    def wrapper(*args: Any, **kwargs: Any):
+        try:
+            return function(*args, **kwargs)
+        except (KeyboardInterrupt, EOFError):
+            print("\nExit repl mode.")
+            sys.exit(0)
+
+    return wrapper
+
+
+@end_repl_mode
 def interact_model(config: Union[str, Path, dict]) -> None:
     """Start interaction with the model described in corresponding configuration file."""
     model = build_model(config)
 
+    print("\nExit repl - type q and press enter, or press ctrl-c, or ctrl-d.")
+
+    def input_data(prompt: str) -> tuple[str]:
+        """Filter and processing input data."""
+        while True:
+            data: str = input(f"\033[34m\033[107m{prompt}:\033[0m ")
+            if data == "":
+                continue
+            if data.isspace():
+                continue
```

Comment:
Remove continue. We should allow user to remain some arguments blank.

Reply:
After removing continue, code from input_data could be moved from separated function back